generated-sample-code/samplegen_basics.js

```diff
@@ -0,0 +1,69 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// DO NOT EDIT! This is a generated sample ("Request",  "samplegen_basics")
+
+// sample-metadata:
+//   title: This is the sample title
+//   description: This is the sample description
+//   usage: node samples/v1/samplegen_basics.js [--display_name "This is the default value of the display_name request field"]
+
+'use strict';
+
+function main(displayName = 'This is the default value of the display_name request field') {
+  // [START samplegen_basics]
+  /**
+  * TODO(developer): Uncomment these variables before running the sample.
+  */
+  // const displayName = 'This is the default value of the display_name request field';
+
+  // Imports the client library
+  const {ProductSearchClient} = require('@google-cloud/vision').v1;
+
+  // Instantiates a client
+  const productSearchClient = new ProductSearchClient();
+
+  async function sampleCreateProductSet() {
+
+    // The project and location in which the product set should be created.
+    const formattedParent = productSearchClient.locationPath('[PROJECT]', '[LOCATION]');
+    const productSet = {
+      displayName: displayName,
+    };
+
+    // Construct request
+    const request = {
+      parent: formattedParent,
+      productSet: productSet,
+    };
+
+    // Run request
+    const [response] = await productSearchClient.createProductSet(request);
```

Comment:
In cases like this, could we do it as a one liner:

```js
const [productSet] = await productSearchClient.createProductSet(request);
```

Reply:
I would love this!

Would you categorize as `nit` or `blocker`?

Do-able fix, but I'd call it a size 👕 Medium, just because `response` is re-used a good bit and we would have to figure out if the type being returned is a resource as opposed to a `FooResponse`. All do-able. I'd love to see this in our backlog.

Reply:
I would say a nit, worth having on your radar though, as in code review I would suggest getting rid of the additional assignment.

Reply:
I would bump the severity of this to a blocker. Users don't know that our samples were auto-generated and don't care in general, they will just think we write weird code :)

Reply:
Ack and that's reasonable! That clarification/distinction is really helpful for prioritizing.